[cmake] Add FindMDNSEmbedded module (for Zeroconf on Android)

fetzerch opened the pull request:
## Description
Zeroconf is not available on androd. It uses mDNSEmbedded which got forgotten when migrating to cmake.

@ping: @MartijnKaijser, @wsnipex, @hudokkow 

## Motivation and Context
Zeroconf for droids.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->
Compile tested.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fetzerch committed:
[cmake] Add FindMDNS module

- Fixes Zeroconf on Android (mDNS_embedded)
- Link Zeroconf on Windows without pragma lib (mDNS)

fetzerch commented:
jenkins build this please

hudokkow commented on project/cmake/modules/FindMDNSEmbedded.cmake:
```diff
@@ -0,0 +1,38 @@
+if(MDNSEMBEDDED_FOUND)
+  set(MDNSEMBEDDED_INCLUDE_DIRS ${MDNSEMBEDDED_INCLUDE_DIR})
+  set(MDNSEMBEDDED_LIBRARIES ${MDNSEMBEDDED_LIBRARY})
+  set(MDNSEMBEDDED_DEFINITIONS -DHAVE_LIBMDNSEMBEDDED=1)
```
From https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules
> Xxx_DEFINITIONS
    Definitions to use when compiling code that uses Xxx. This really shouldn’t include options such as -DHAS_JPEG that a client source-code file uses to decide whether to #include <jpeg.h>

Is it really needed?
Otherwise looks good.

fetzerch commented:
Unfortunately there are some undefined symbols now on android which got introduced by https://github.com/xbmc/xbmc/pull/10025. Guess that just slipped in because we didn't use the lib on android since we moved to cmake. Any idea how to fix or whom to ping? (@Rechi, @Memphiz.)

```
[100%] Linking CXX shared library libkodi.so
mDNSCore/mDNS.c:8688: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:4386: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:11945: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:13157: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:13137: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:13145: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:7271: error: undefined reference to 'InitializeAnonInfoForQuestion'
mDNSCore/mDNS.c:7318: error: undefined reference to 'SetAnonData'
mDNSCore/mDNS.c:9159: error: undefined reference to 'InitializeAnonInfoForCR'
mDNSCore/mDNS.c:8544: error: undefined reference to 'IdenticalAnonInfo'
mDNSCore/mDNS.c:9232: error: undefined reference to 'CopyAnonInfoForCR'
mDNSCore/mDNS.c:12069: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:12957: error: undefined reference to 'ReInitAnonInfo'
mDNSCore/mDNS.c:12970: error: undefined reference to 'ReInitAnonInfo'
mDNSCore/DNSCommon.c:1907: error: undefined reference to 'AnonInfoAnswersQuestion'
mDNSCore/DNSCommon.c:1941: error: undefined reference to 'AnonInfoAnswersQuestion'
mDNSCore/DNSCommon.c:1820: error: undefined reference to 'AnonInfoAnswersQuestion'
```

I'll fix the compile errors on non droid platforms (probably there's some wrong with conditionally compiling it in).

Rechi commented:
@fetzerch maybe https://github.com/Rechi/xbmc/commit/10feb8e16feaaf03fa2201dfb7c23e88767910db fixes the issue, but I have no android built environment to test.

fetzerch committed:
[depends] Fix undefined symbols on mdnsresponder

Introduced by https://github.com/xbmc/xbmc/pull/10025.

mDNSCore/mDNS.c:8688: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:4386: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:11945: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:13157: error: undefined reference to 'FreeAnonInfo'
mDNSCore/mDNS.c:13137: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:13145: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:7271: error: undefined reference to 'InitializeAnonInfoForQuestion'
mDNSCore/mDNS.c:7318: error: undefined reference to 'SetAnonData'
mDNSCore/mDNS.c:9159: error: undefined reference to 'InitializeAnonInfoForCR'
mDNSCore/mDNS.c:8544: error: undefined reference to 'IdenticalAnonInfo'
mDNSCore/mDNS.c:9232: error: undefined reference to 'CopyAnonInfoForCR'
mDNSCore/mDNS.c:12069: error: undefined reference to 'AllocateAnonInfo'
mDNSCore/mDNS.c:12957: error: undefined reference to 'ReInitAnonInfo'
mDNSCore/mDNS.c:12970: error: undefined reference to 'ReInitAnonInfo'
mDNSCore/DNSCommon.c:1907: error: undefined reference to 'AnonInfoAnswersQuestion'
mDNSCore/DNSCommon.c:1941: error: undefined reference to 'AnonInfoAnswersQuestion'
mDNSCore/DNSCommon.c:1820: error: undefined reference to 'AnonInfoAnswersQuestion'

fetzerch commented:
jenkins build this please

Memphiz commented:
What do you mean you didn't use the lib since the cmake move? Zeroconf was broken for android since then?

fetzerch commented:
Yep. Seems no one noticed until yesterday.

fetzerch commented:
jenkins build this please

fetzerch commented:
Builds are fine now.
@MartijnKaijser: Could you give http://mirrors.kodi.tv/test-builds/android/arm/kodi-20161222-f22c6ec-cmake_mdns-armeabi-v7a.apk a try please?

fetzerch commented:
> @fetzerch maybe Rechi@10feb8e fixes the issue, but I have no android built environment to test.

@Rechi thanks, that seems to do the trick

MartijnKaijser commented:
@fetzerch seems to work.
